utils/sim_configuration/generateImsi.test.ts

```diff
@@ -0,0 +1,49 @@
+import { generateUniqueImsi } from "@/utils/sim_configuration/generateImsi";
+
+
+describe('IMSI Generator Unit Tests', () => {
+    let existingIMSIs: Set<string>;
+    beforeEach(() => {
+        // Initialize an empty set of existing IMSIs for each test
+        // In the implementation, getAllImsis from @utils/callSubscriberApi should be called to get existingIMSIs.
+        existingIMSIs = new Set();
+    });
+    it("should generate a unique IMSI with valid MCC, MNC", () => {
+        const mcc = "310";
+        const mnc = "410";
+
+        const imsi = generateUniqueImsi(mcc, mnc, existingIMSIs);
+
+        // Validate that the generated IMSI starts with MCC + MNC and is 16 digits long
+        expect(imsi).toMatch(new RegExp(`^${mcc}${mnc}\\d{10}$`));
+        // IMSI length should be 16.
+        expect([16]).toContain(imsi.length);
```

Comment:
```suggestion
        expect(imsi.length).toEqual(16);
```
something like this is more readable

Reply:
Done.